src/org/lsst/ts/jenkins/components/Csc.groovy

```diff
@@ -332,23 +332,26 @@ def update_container_branches() {
     withEnv([]){
         sh """
         set +x
+	shopt -s extglob
         source /home/saluser/.setup_dev.sh || echo loading env failed. Continuing...
 
         for repo in \$(ls /home/saluser/repos/)
         do
             cd /home/saluser/repos/\$repo
             /home/saluser/.checkout_repo.sh ${WORK_BRANCHES}
-            git pull
         done
+	cd ${env.WORKSPACE}/ci/
+	# Deal with some file oddities
+	rm Jenkinsfile || true
+	rm -rf *@tmp*
+	ls ${env.WORKSPACE}/ci/
         for repo in \$(ls ${env.WORKSPACE}/ci/)
         do
-            cd ${env.WORKSPACE}/ci/\$repo
-            git_branch=\$(git rev-parse --abbrev-ref HEAD)
+	    echo \$repo
+	    cd ${env.WORKSPACE}/ci/\$repo
+	    git_branch=\$(git rev-parse --abbrev-ref HEAD)
             git branch --set-upstream-to=origin/\$git_branch \$git_branch
             /home/saluser/.checkout_repo.sh ${WORK_BRANCHES}
-            git pull
-            eups declare -r . -t current
```

Comment:
I understand why you removed `setup -kr .` but am confused by removing `eups declare...` (and, `git pull`). Where are you eups declaring the packages now?

Reply:
Eric explained it to me. eups declare now happens in an earlier phase, and git pull was never needed.